// permissionless/_cjs/clients/decorators/bundler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bundlerActions = void 0;
const chainId_js_1 = require("../../actions/bundler/chainId.js");
const estimateUserOperationGas_js_1 = require("../../actions/bundler/estimateUserOperationGas.js");
const getUserOperationByHash_js_1 = require("../../actions/bundler/getUserOperationByHash.js");
const getUserOperationReceipt_js_1 = require("../../actions/bundler/getUserOperationReceipt.js");
const sendUserOperation_js_1 = require("../../actions/bundler/sendUserOperation.js");
const supportedEntryPoints_js_1 = require("../../actions/bundler/supportedEntryPoints.js");
const waitForUserOperationReceipt_js_1 = require("../../actions/bundler/waitForUserOperationReceipt.js");
const bundlerActions = (client) => ({
    sendUserOperation: async (args) => (0, sendUserOperation_js_1.sendUserOperation)(client, args),
    estimateUserOperationGas: (args) => (0, estimateUserOperationGas_js_1.estimateUserOperationGas)(client, args),
    supportedEntryPoints: () => (0, supportedEntryPoints_js_1.supportedEntryPoints)(client),
    chainId: () => (0, chainId_js_1.chainId)(client),
    getUserOperationByHash: (args) => (0, getUserOperationByHash_js_1.getUserOperationByHash)(client, args),
    getUserOperationReceipt: (args) => (0, getUserOperationReceipt_js_1.getUserOperationReceipt)(client, args),
    waitForUserOperationReceipt: (args) => (0, waitForUserOperationReceipt_js_1.waitForUserOperationReceipt)(client, args)
});
exports.bundlerActions = bundlerActions;
