// permissionless/_cjs/actions/smartAccount/sendTransactions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendTransactions = void 0;
const getAction_js_1 = require("../../utils/getAction.js");
const index_js_1 = require("../../utils/index.js");
const waitForUserOperationReceipt_js_1 = require("../bundler/waitForUserOperationReceipt.js");
const sendUserOperation_js_1 = require("./sendUserOperation.js");
async function sendTransactions(client, args) {
    const { account: account_ = client.account, transactions, sponsorUserOperation, maxFeePerGas, maxPriorityFeePerGas, nonce } = args;
    if (!account_) {
        throw new index_js_1.AccountOrClientNotFoundError({
            docsPath: "/docs/actions/wallet/sendTransaction"
        });
    }
    const account = (0, index_js_1.parseAccount)(account_);
    if (account.type !== "local") {
        throw new Error("RPC account type not supported");
    }
    const callData = await account.encodeCallData(transactions.map(({ to, value, data }) => {
        if (!to)
            throw new Error("Missing to address");
        return {
            to,
            value: value || 0n,
            data: data || "0x"
        };
    }));
    const userOpHash = await (0, getAction_js_1.getAction)(client, sendUserOperation_js_1.sendUserOperation)({
        userOperation: {
            sender: account.address,
            paymasterAndData: "0x",
            maxFeePerGas: maxFeePerGas || 0n,
            maxPriorityFeePerGas: maxPriorityFeePerGas || 0n,
            callData: callData,
            nonce: nonce
        },
        account: account,
        sponsorUserOperation
    });
    const userOperationReceipt = await (0, getAction_js_1.getAction)(client, waitForUserOperationReceipt_js_1.waitForUserOperationReceipt)({
        hash: userOpHash
    });
    return userOperationReceipt?.receipt.transactionHash;
}
exports.sendTransactions = sendTransactions;
