// permissionless/_esm/accounts/privateKeyToSafeSmartAccount.js
import {} from "viem";
import { privateKeyToAccount } from "viem/accounts";
import { signerToSafeSmartAccount } from "./signerToSafeSmartAccount.js";
/**
 * @description Creates an Simple Account from a private key.
 *
 * @returns A Private Key Simple Account.
 */
export async function privateKeyToSafeSmartAccount(client, { privateKey, safeVersion, entryPoint, addModuleLibAddress: _addModuleLibAddress, safe4337ModuleAddress: _safe4337ModuleAddress, safeProxyFactoryAddress: _safeProxyFactoryAddress, safeSingletonAddress: _safeSingletonAddress, multiSendAddress: _multiSendAddress, multiSendCallOnlyAddress: _multiSendCallOnlyAddress, saltNonce = 0n, safeModules = [], setupTransactions = [] }) {
    const privateKeyAccount = privateKeyToAccount(privateKey);
    return signerToSafeSmartAccount(client, {
        signer: privateKeyAccount,
        safeVersion,
        entryPoint,
        addModuleLibAddress: _addModuleLibAddress,
        safe4337ModuleAddress: _safe4337ModuleAddress,
        safeProxyFactoryAddress: _safeProxyFactoryAddress,
        safeSingletonAddress: _safeSingletonAddress,
        multiSendAddress: _multiSendAddress,
        multiSendCallOnlyAddress: _multiSendCallOnlyAddress,
        saltNonce,
        safeModules,
        setupTransactions
    });
}
