// permissionless/_cjs/accounts/privateKeyToSafeSmartAccount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.privateKeyToSafeSmartAccount = void 0;
const accounts_1 = require("viem/accounts");
const signerToSafeSmartAccount_js_1 = require("./signerToSafeSmartAccount.js");
async function privateKeyToSafeSmartAccount(client, { privateKey, safeVersion, entryPoint, addModuleLibAddress: _addModuleLibAddress, safe4337ModuleAddress: _safe4337ModuleAddress, safeProxyFactoryAddress: _safeProxyFactoryAddress, safeSingletonAddress: _safeSingletonAddress, multiSendAddress: _multiSendAddress, multiSendCallOnlyAddress: _multiSendCallOnlyAddress, saltNonce = 0n, safeModules = [], setupTransactions = [] }) {
    const privateKeyAccount = (0, accounts_1.privateKeyToAccount)(privateKey);
    return (0, signerToSafeSmartAccount_js_1.signerToSafeSmartAccount)(client, {
        signer: privateKeyAccount,
        safeVersion,
        entryPoint,
        addModuleLibAddress: _addModuleLibAddress,
        safe4337ModuleAddress: _safe4337ModuleAddress,
        safeProxyFactoryAddress: _safeProxyFactoryAddress,
        safeSingletonAddress: _safeSingletonAddress,
        multiSendAddress: _multiSendAddress,
        multiSendCallOnlyAddress: _multiSendCallOnlyAddress,
        saltNonce,
        safeModules,
        setupTransactions
    });
}
exports.privateKeyToSafeSmartAccount = privateKeyToSafeSmartAccount;
